BraveWallet/Crypto/Transaction Confirmations/EditNonceView.swift

```diff
@@ -0,0 +1,79 @@
+// Copyright 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import SwiftUI
+import BraveUI
+import BraveCore
+import struct Shared.Strings
+
+struct EditNonceView: View {
+  var confirmationStore: TransactionConfirmationStore
+  var transaction: BraveWallet.TransactionInfo
+  
+  @Environment(\.presentationMode) @Binding private var presentationMode
+  @State private var nonce = ""
+  
+  var body: some View {
+    List {
+      Section(
+        header: Text(Strings.Wallet.editNonceHeader)
+          .textCase(.none),
+        footer: Text(Strings.Wallet.editNonceFooter)
+      ) {
+        TextField(Strings.Wallet.editNoncePlaceholder, text: $nonce)
+          .keyboardType(.numberPad)
+      }
+      Section {
+        Button(action: {
+          if let value = Int(nonce) {
+            let nonceHex = "0x\(String(format: "%02x", value))"
+            confirmationStore.editNonce(
+              for: transaction,
+              nonce: nonceHex) { success in
+                if success {
+                  presentationMode.dismiss()
+                } else {
+                  // Show error?
```

Comment:
Probably should show an error then lol?

Reply:
@nuo-xu will open a new issue to handle error states since this matches whats shipping for edit gas fee/priority fee

Reply:
https://github.com/brave/brave-ios/issues/5081 for adding error handling